ibm/service/power/resource_ibm_pi_spp_placement_group.go

```diff
@@ -0,0 +1,152 @@
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			Arg_SPPPlacementGroupName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the SPP placement group",
+			},
+
+			Arg_SPPPlacementGroupPolicy: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"affinity", "anti-affinity"}),
+				Description:  "Policy of the SPP placement group",
+			},
+
+			helpers.PICloudInstanceId: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PI cloud instance ID",
+			},
+
+			Attr_SPPPlacementGroupMembers: {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Member SPP IDs that are the SPP placement group members",
+			},
+
+			Attr_SPPPlacementGroupID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "SPP placement group ID",
+			},
+		},
+	}
+}
+
+func resourceIBMPISPPPlacementGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(helpers.PICloudInstanceId).(string)
+	name := d.Get(Arg_SPPPlacementGroupName).(string)
+	policy := d.Get(Arg_SPPPlacementGroupPolicy).(string)
+	client := st.NewIBMPISPPPlacementGroupClient(ctx, sess, cloudInstanceID)
+	body := &models.SPPPlacementGroupCreate{
+		Name:   &name,
+		Policy: &policy,
+	}
+
+	response, err := client.Create(body)
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.FromErr(err)
+	}
+
+	log.Printf("Printing the SPP placement group %+v", &response)
+
+	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, *response.ID))
+	return resourceIBMPISPPPlacementGroupRead(ctx, d, meta)
+}
+
+func resourceIBMPISPPPlacementGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := parts[0]
+	client := st.NewIBMPISPPPlacementGroupClient(ctx, sess, cloudInstanceID)
+
+	response, err := client.Get(parts[1])
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.FromErr(err)
+	}
+
+	d.Set(Arg_SPPPlacementGroupName, response.Name)
+	d.Set(Attr_SPPPlacementGroupID, response.ID)
+	d.Set(Arg_SPPPlacementGroupPolicy, response.Policy)
+	d.Set(Attr_SPPPlacementGroupMembers, response.MemberSharedProcessorPools)
+
+	return nil
+
+}
+
+func resourceIBMPISPPPlacementGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
if no update is present you can remove this method and definitation also from the schema

Reply:
Done